src/components/SideBar/SideBar.jsx

```diff
@@ -1,2 +1,83 @@
-import React from "react";
-export const SideBar = () => <>SideBar</>;
+import { useState, useRef, useEffect } from "react";
+import {
+  StyledText,
+  StyledSidebar,
+  StyledLink,
+  StyledTopWrapper,
+  StyledLogoSvg,
+  StyledLogoText,
+  StyledBarsOutlined,
+  StyledSvgWrapper,
+  StyledSideBarContainer,
+  StyledInputWrapper,
+  StyledSearch,
+  StyledSearchBtn,
+} from "./style";
+import { menuItemsArr } from "./menuItems";
+import { SearchOutlined } from "@ant-design/icons";
+
+export const SideBar = () => {
+  const [isOpen, setIsOpen] = useState(false);
+  const toggleSidebar = () => setIsOpen(!isOpen);
+  const menuItems = menuItemsArr;
+  const sidebarRef = useRef(null);
+
+  const handleClickOutside = (event) => {
+    if (sidebarRef.current && !sidebarRef.current.contains(event.target)) {
```

Comment:
Instead of creating a ref and checking it you can just add handler for wrapping component and stop event propagation. With that your event would not bubble higher and though will not trigger onclick on document